testing-commandments.md

```diff
@@ -0,0 +1,30 @@
+# 8 Commandments of Automated Testing
+
+*DISCLAIMER* - Failure to meet any of these points doesn't mean new ideas or pull requests should be outright rejected. They should be used as points of discussion to ultimately help us ship code faster.
+
+`code` is used in reference to the code being tested. The target, maybe?
+`test` is used in reference to the code that is exercising the codebase to validate correctness.
+
+## Thou shalt be consistent within projects
+If a project has a de-facto standard, it should be maintained. The process for changing a de-facto standard should be people focused, and should not be initially communicated via a pull request.
+
+## Thy tests should be obvious
+If the test behaviour isn't obvious by looking at it, it's no good. If I have to scroll for days to read a test, it's no good.
+
+## Thy tests should only check for something once
+If I make a single code change and multiple tests fail, it's no good.
+
+## Broken code should break thy tests
+If I can delete code and it's corresponding test suite still passes, it's no good.
```

Comment:
I think we're all agreeing with what is written here? Given a test file `A` and a file full of code `B`, if I go in and delete code from `B`, I expect the test suite to fail. It makes it clear that we want people to write tests to cover their core functionality to prevent other developers from accidentally breaking stuff, it compliments the last commandment (`Thy tests should be described via emergent properties`), and it also leaves the "how" open to creative thinking.